v8/usr/man/man1/monk.1
.TH MONK 1
.SH NAME
monk, monksample \- typeset documents and letters
.SH SYNOPSIS
.B monk
[ options ] [ files ]
.PP
.B monksample
[ sample ]
.SH DESCRIPTION
.I Monk
formats the text in the named
.I files
for
phototypesetting, using other
.IR troff (1)
preprocessors as necessary.
.I Options\^
are given below.
Any other arguments or flags (e.g.,
.BR \-o1\-2 )
are passed
to
.IR troff (1).
Options can occur in any order and can be intermixed with files.
If no arguments are given,
.I monk
prints a list of its options.
.PP
.PD 0
.TP
. \".B \-12
. \"Indicates that the document is to be produced in 12-pitch.
. \"May be used with
. \".B $TERM
. \"is set to one of
. \".BR 300 ,
. \".BR 300s ,
. \".BR 450 ,
. \"and
. \".BR 1620 .
. \"(The pitch switch on the
. \".SM DASI
. \"300 and 300s terminals must be manually set to
. \".B 12
. \"if this option is used.)
. \".TP
.BI \-A commands
Invoke the
.I commands
after all preprocessors and before
.IR troff (1).
.TP
.BI \-B commands
Invoke the
.I commands
after
.I monk
and before any other preprocessor.
.TP
.B \-E
Invoke the
.B \-e
option of
.IR troff (1).
.TP
.BR \-R file
Use
.I file
as the index file for
.IR refer (1).
.TP
.BI \-T dest
Prepare output for typesetter or terminal
.IR dest ;
Where dest can be:
.ta \w'\-T52imagen  'u
.br
.BR \-T202 "	Mergenthaler Linotron 202"
.br
.BR \-T9700 "	Xerox 9700 printer"
.br
.BR \-Taps "	Autologic APS-5"
.br
.BR \-Tcanon "	Canon laser printer, see"
.IR candest (7)
(default)
.br
.BR \-Tcat "	Graphic Systems C/A/T"
.br
.BR \-Timagen "	Imagen printer, see"
.IR i10send (1)
.br
.BR \-Tjerq "	5620 or blit terminal"
. \".br
. \".BR \-Tlp "	line printer"
. \".br
. \"If this option is
. \".I not
. \"used,
. \".I monk
. \"will use the shell variable
. \".B $TERM
. \"from the environment (See
. \".IR profile (5)
. \"and
. \".IR environment (7))
. \"as the value of
. \".IR dest .
.TP
.B \-T\-
The output from
.IR troff (1)
is placed on standard output instead of the default printer.
.TP
.B \-x
Shows the preprocessors that are being invoked.
.PD
.PP
The following options are not normally needed because monk automatically
determines which preprocessor are required.
However, if the commands in
.B \-A
or
.B \-B
options require a preprocessor, you can force
.I monk
to invoke them using the following options.
.PP
.PD 0
.TP
.B \-c
.IR col (1)
(postprocessor)
. \"note that
. \".IR col (1)
. \"is invoked automatically by
. \".I monk
. \"for the
. \".BR \-T300 ,
. \".BR \-T300s ,
. \".BR \-T450 ,
. \".BR \-T4014 ,
. \".BR \-Ttek ,
. \".BR \-T1620 ,
. \".BR \-T300-12 ,
. \".BR \-T300s-12 ,
. \".BR \-T450-12 ,
. \".BR \-T1620-12 ,
. \".BR \-T37 ,
. \".BR \-T4000a ,
. \".BR \-T382 ,
. \".BR \-TX ,
. \".BR \-Thp ,
. \".BR \-T2621 ,
. \".BR \-T2640 ,
. \".BR \-T2645 ,
. \".BR \T735 ,
. \".BR \-T745 ,
. \".BR \-T43 ,
. \".BR \-T40/4 ,
. \".BR \-T40/2
. \"terminals.
.TP
.B \-cn
.IR cite (1)
.TP
.B \-e
.IR eqn (1)
.TP
.B \-g
.IR grap (1)
.TP
.B \-i
.IR ideal (1)
.TP
.B \-p
.IR pic (1)
.TP
.B \-r
.IR refer (1)
.TP
.B \-s
.IR soelim (1)
.TP
.B \-t
.IR tbl (1)
.TP
.B \-tp
.IR tped (1)
.PD
.PP
.I Monksample
produces on the standard output a skeleton document that you can
redirect into a file and edit.
If no argument is given,
.I monksample
prints a list of the available
.I samples.
They are:
.PP
.PD 0
.TP 8
.B cspress
Computer Science Press galley sheets.
.TP
.B letter
Personal letter
.TP
.B model
IEEE/ACM model sheets.
.TP
.B rp
AT&T Bell Laboratories release paper.
.TP
.B song
Song sheets for singing at nursing homes.
.TP
.B tm
AT&T Bell Laboratories technical memorandum.
.PD
.SH FILES
.ta \w'/usr/lib/monk/dorefer  'u
.br
/usr/lib/monk/cite	forward and backward reference preprocessor
.br
/usr/lib/monk/db	monk databases
.br
/usr/lib/monk/dorefer	sed script for
.IR refer (1)
.br
/usr/lib/monk/monk	monk compiler
.br
/usr/lib/monk/soelim	preprocessor for
.IR troff (1)
`.so' directives
.br
/usr/lib/monk/tmac.p	macros for
.IR pic (1)
preprocessor
.br
/usr/lib/monk/sample	directory for existing samples
.SH "SEE ALSO"
Murrel, S. L.,
Kowalski, T. J.,
.I "Typing Documents on the UNIX System: Using Monk 0.2"
.br
apsend(1),
cite(1),
d202(1),
. \"environment(7),
eqn(1),
grap(1),
ideal(1),
ped(9.1),
pic(1),
. \"profile(5),
proof(9.1),
refer(1),
tbl(1),
troff(1)


v8/usr/src/cmd/monk/cmd/monk.sh

#
# initalizations
#
arg0=tmonk

if [ "$CANDEST" ]
then
	candest="CANDEST=$CANDEST"
fi

if [ "$I10DEST" ]
then
	i10dest="I10DEST=$I10DEST"
fi

if [ $arg0 = tmonk ]
then
	case `uuname -l` in
	
	mhuxm)
		MHOME=/m2/c1122/frodo/lib/monk
		STDIN=-
		TERM=-Ti10
		eqn='eqn'
		i10send='i10send'
		nulloutput=
		output="|$i10dest i10send -von"
		roff='troff'
		;;
	
	alice)
		STDIN=/dev/stdin
		TERM=-Ti10
		eqn='eqn'
		i10send='/usr/frodo/bin/ui10send'
		nulloutput=
		output="|$i10dest /usr/frodo/bin/ui10send -von"
		roff='troff'
		;;
	
	araki)
		STDIN=/dev/stdin
		TERM=-Ti10
		eqn='eqn'
		i10send='ui10send'
		nulloutput=
		output='|ui10send -d5211out -von'
		roff='troff'
		;;
	
	*)
		STDIN=/dev/stdin
		TERM=-T202
		eqn='eqn'
		i10send='ui10send'
		nulloutput=
		output="|$candest dcan"
		roff='troff'
		;;
	
	esac

else
	STDIN=-
	TERM=-T${TERM-450}
	eqn='neqn'
	nulloutput=
	output=
	roff='nroff'
fi

#
# path names for macros and default references
#
MHOME=${MHOME-/usr/lib/monk}

MDB=${MDB-$MHOME/db}

EQN=/usr/pub/eqnchar

MP=${TMAC-$MHOME}/tmac.p

REFS=${REFS-/usr/frodo/refs/index}

PATH=$MHOME:$PATH
export PATH

#
# get options
#
for i
do
	case "$i" in

#
# troff, reference or output options
#
	-12)
		pitch=-12
		;;

	-A*)
		Acmd="|"`expr "$i" : '..\(.*\)'`
		;;

	-B*)
		Bcmd="|"`expr "$i" : '..\(.*\)'`
		;;

	-E)
		options="$options -e"
		;;

	-R*)
		REFS=`expr "$i" : '..\(.*\)'`
		;;

	-T202)
		TERM=$i
		output='|d202'
		;;

	-T9700)
		TERM=$i
		output='|dpr -mx'
		;;

	-Taps)
		TERM=$i
		output='|apsend'
		;;

	-Tcanon)
		i=-T202
		TERM=$i
		output="|$candest dcan"
		;;

	-Tcat)
		TERM=$i
		output='|gcat -f "$files"'
		;;

	-Timagen)
		i=-Ti10
		TERM=$i
		output="|$i10dest $i10send -von"
		;;

	-Tjerq)
		i=-T202
		TERM=$i
		output="|proof; cat /tmp/e$$; rm -f /tmp/e$$"
		exec 2> /tmp/e$$
		;;

	-Tlp)
		TERM=$i
		output='|lpr'
		;;

	-T-)
		output=$nulloutput
		;;

	-T*)
		TERM=$i
		;;

	-a)
		options="$options $i"
		roff=troff
		output=
		;;

	-x)
		verbose=on
		;;

	-*)
		preprocessors="$preprocessors $i"
		;;

	*)
		files="$files $i"
		;;

	esac
done

#
# find what pre processors monk needs
#
preprocessors=`
egrep -h '\|(source|remember|reference|equation|e|table|picture|ideal|ped|graph)[(<[{'\''"\`]|\|begin[(<[{'\''"\`](source|remember|reference|equation|e|table|picture|ideal|ped|graph)' $files |
awk '
/\|source[(<[{'\''"\`]/		{soelim++}
/\|begin[(<[{'\''"\`]source/	{soelim++}
/\|remember[(<[{'\''"\`]/	{cite++}
/\|begin[(<[{'\''"\`]remember/	{cite++}
/\|reference[(<[{'\''"\`]/	{refer++}
/\|begin[(<[{'\''"\`]reference/	{refer++}
/\|equation[(<[{'\''"\`]/	{eqn++}
/\|begin[(<[{'\''"\`]equation/	{eqn++}
/\|e[(<[{'\''"\`]/		{eqn++}
/\|begin[(<[{'\''"\`]e/		{eqn++}
/\|table[(<[{'\''"\`]/		{tbl++}
/\|begin[(<[{'\''"\`]table/	{tbl++}
/\|picture[(<[{'\''"\`]/	{pic++}
/\|begin[(<[{'\''"\`]picture/	{pic++}
/\|ideal[(<[{'\''"\`]/		{ideal++}
/\|begin[(<[{'\''"\`]ideal/	{ideal++}
/\|ped[(<[{'\''"\`]/		{tped++}
/\|begin[(<[{'\''"\`]ped/	{tped++}
/\|graph[(<[{'\''"\`]/		{grap++; pic++}
/\|begin[(<[{'\''"\`]graph/	{grap++; pic++}
END 				{
				 x = ""
				 if (soelim)  x = x " -s"
				 if (refer) x = x " -r"
				 if (cite)  x = x " -cn"
				 if (ideal) x = x " -i"
				 if (grap)  x = x " -g"
				 if (pic)   x = x " -p"
				 if (tped)  x = x " -tp"
				 if (tbl)   x = x " -t"
				 if (eqn)   x = x " -e"
				 print x
				}
'`" $preprocessors"

#
# get preprocessors and other options
#
for i in $preprocessors
do
	case "$i" in

#
# pre-processors
#
	-c)
		ccmd='|col'
		;;

	-cn)
		cncmd='|cite'
		erroutput='2>> .cite'
		;;

	-e)
		ecmd="|$eqn"
		;;

	-g)
		gcmd='|grap'
		;;

	-i)	icmd='|ideal -q'
		;;

	-p)
		pcmd='|pic'
		;;

	-r)
		rcmd="|refer -e -n -p $REFS |sed -f $MHOME/dorefer"
		;;

	-s)
		scmd='|soelim /dev/stdin'
		;;

	-t)
		tcmd='|tbl'
		;;

	-tp)
		tpcmd='|tped'
		;;

	-*)
		options="$options $i"
		;;

	esac
done


#
# special handeling for terminals
#
	if [ "$pitch" ]
	then
		case $TERM in

		-T300|-T300s|-T450|-T1620)
			TERM="$TERM-12"
			;;
		esac
	fi

	case $arg0  in

	tmonk)
		case $TERM in

		-Tcat|-Taps|-Ti10|-T202)
			;;


		-Tdover)
			TERM=
			;;

		*)
			echo 1>&2 Invalid $TERM
			exit 1
			;;

		esac
		;;

	nmonk)
		if [ "ccmd" ]
		then
			case $TERM in

			-T300|-T300s|-T450|-T4014|-Ttek|-T1620)
				ccmd="|col -f|greek $TERM"
				;;

			-T300-12|-T300s-12|-T450-12|-T1620-12)
				ccmd='|col -f|greek $TERM'
				options='$options -rW72'
				;;

			-T37|-T4000a|-T382|-TX)
				ccmd='|col -f'
				;;

			-Thp|-T2621|-T2640|-T2645)
				ccmd='|colcrt -'
				;;

			-T735|-T745)
				ccmd='|col -x'
				;;

			-T43)
				ccmd='|col -x'
				options='$options -rW75'
				;;

			-T40/4|-T40/2)
				ccmd='|col -b'
				;;

			esac
		fi

		case $TERM in

		-T1620)
			TERM=-T450
			;;

		-T1620-12)
			TERM=-T450-12
			;;

		-T37|-Ttn300|-T300S|-T300*|-T450*)
			;;

		*)
			TERM=
			;;

		esac
		;;

	*)
		echo 1>&2 This command must be called nmonk or troff
		exit 1
		;;

	esac

#
# special handleing for pic and eqn
#
if [ "$pcmd" ]
then
	macro="$macro $MP"

	case $TERM in

	-Ti10)
		pcmd="$pcmd -T240"
		;;

	-T202)
		pcmd="$pcmd -D"
		;;

	*)
		pcmd="$pcmd $TERM"
		;;

	esac
fi

if [ "$ecmd" ]
then
	case $TERM in

	-Ti10)
		ecmd="$ecmd -r240 -m6"
		;;

	-Taps)
		ecmd="$ecmd -Taps"
		;;

	-T202)
		ecmd="$ecmd -T202"
		;;

	esac

	ecmd="$ecmd -d $EQN $STDIN"
fi

#
# use default terminal for 202 so pic won't get confused with -T202
#
if [ "$TERM" = -T202 ]
then
	TERM=
fi

#
# do it
#
if [ $verbose ]
then
	echo 1>&2 "$MHOME/monk -d $MDB $files | sed -f $MHOME/dokludge $Bcmd \\
	$scmd $cncmd $rcmd $icmd $gcmd $tcmd $pcmd $tpcmd $ecmd $Acmd \\
	| $roff $TERM $options $macro - $erroutput $ccmd $output"
fi
exec sh -c "exec $MHOME/monk -d $MDB $files | sed -f $MHOME/dokludge $Bcmd \\
	$scmd $cncmd $rcmd $icmd $gcmd $tcmd $pcmd $tpcmd $ecmd $Acmd \\
	| $roff $TERM $options $macro - $erroutput $ccmd $output"
